### PR TITLE
SIDM-9210 naming pattern changed

### PR DESCRIPTION
**Pull request by arun-hmcts**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/test/js/shared/idam_helper.js [hunk 1]
```diff
@@ -845,6 +845,150 @@ class IdamHelper extends Helper {
...
+            TestData.FUNCTIONAL_TEST_TOKEN = tokenData.access_token;
+            return tokenData.access_token; // Assuming token is in the 'access_token' field
+        } catch (error) {
+            console.trace('Error fetching token:', error);
+            throw error;
+        }
+    }
+
+    createServiceUsingTestingSupportService(serviceName, serviceClientSecret, roleId, token, scope = [], ssoProviders = '') {
+        let data;
```
Comment (jburke-idam) [inline src/test/js/shared/idam_helper.js:884]: testing support api uses role names, not ids

File: src/test/js/shared/idam_helper.js [hunk 2]
```diff
@@ -845,6 +845,150 @@ class IdamHelper extends Helper {
...
+                    refreshTokenLifetime: "PT0S"
+                }};
+
+        } else {
+
+            data = {
+                clientId: serviceName,
+                clientSecret : serviceClientSecret,
+                description: serviceName,
+                allowedRoles: [roleId, 'auto-admin_role'],
```
Comment (jburke-idam) [inline src/test/js/shared/idam_helper.js:920]: allowedRoles isn't an attribute on the v2 model: https://github.com/hmcts/idam-api/blob/master/idam-api/src/main/java/uk/gov/hmcts/reform/idam/api/v2/model/ServiceProvider.java

Presumably the roleId here is being used as an onboarding role

File: src/test/js/shared/idam_helper.js [hunk 3]
```diff
@@ -845,6 +845,150 @@ class IdamHelper extends Helper {
...
+        } catch (error) {
+            console.trace('Error fetching token:', error);
+            throw error;
+        }
+    }
+
+    createServiceUsingTestingSupportService(serviceName, serviceClientSecret, roleId, token, scope = [], ssoProviders = '') {
+        let data;
+
+        if (roleId === '') {
```
Comment (jburke-idam) [inline src/test/js/shared/idam_helper.js:886]: Don't like the amount of duplication here. Need to figure out what roleId is being used for in tests (see other comments - it's probably for private beta testing), and build the service object up as one set of code.

File: src/test/js/shared/idam_helper.js [hunk 4]
```diff
@@ -845,6 +845,150 @@ class IdamHelper extends Helper {
...
+                    ],
+                    scopes: scope,
+                    redirectUris: [TestData.SERVICE_REDIRECT_URI],
+                    accessTokenLifetime: "PT0S",
+                    refreshTokenLifetime: "PT0S"
+                }
+            };
+
+        }
+        console.trace(JSON.stringify(data));
```
Comment (jburke-idam) [inline src/test/js/shared/idam_helper.js:945]: if you're going to leave logging in then you probably need to add some TODOs to tidy this up later on.

File: src/test/js/shared/idam_helper.js [hunk 5]
```diff
@@ -845,6 +845,150 @@ class IdamHelper extends Helper {
...
+            }
+            return response.json();
+        }).catch(err => {
+            console.log(err);
+        });
+    }
+
+    createRoleUsingTestingSupportService(roleName, roleDescription, assignableRoles, api_auth_token) {
+        const roleId = uuid.v4();
+        const data = {
```
Comment (jburke-idam) [inline src/test/js/shared/idam_helper.js:966]: doesn't match the V2 model: https://github.com/hmcts/idam-api/blob/master/idam-api/src/main/java/uk/gov/hmcts/reform/idam/api/v2/model/Role.java

Comment (jburke-idam) [reply src/test/js/shared/idam_helper.js:966]: I'm actually quite surprised that this works - I guess that means the assignable roles aren't used for anything in web public (or aren't used in the tests you've changed)

File: src/test/js/shared/idam_helper.js (hunk 2 above)
Comment (arun-hmcts) [reply src/test/js/shared/idam_helper.js]: Removed it now

File: src/test/js/shared/idam_helper.js [hunk 6]
```diff
@@ -845,6 +867,130 @@ class IdamHelper extends Helper {
...
+            console.error("Error:", err);
+            throw err;
+        });
+    }
+
+    async  createRoleUsingTestingSupportService(roleName, roleDescription, assignableRoles, token) {
+        try {
+            const roleId = uuid.v4();
+            const data = {
+                assignableRoles: assignableRoles,
```
Comment (jburke-idam) [inline src/test/js/shared/idam_helper.js:968]:
```suggestion
                assignableRoleNames: assignableRoles,
```

**Review by chandkanuri — APPROVED** [review]

File: src/test/js/shared/idam_helper.js [hunk 7]
```diff
@@ -346,13 +217,15 @@ class IdamHelper extends Helper {
...
-            .catch(err => {
-                console.log(err)
-            });
+        }).then(res => {
+            console.log("****createUserUsingTestingSupportService :*****"+res.status); // Log the response data
+            return res.json();
+        }).then(json => {
+            return json;
+        }).catch(err => {
+            console.error(err); // Log any errors that occur during the request
```
[the remark this answers is not in the document]
Comment (arun-hmcts) [reply src/test/js/shared/idam_helper.js]: Resolved

File: src/test/js/shared/random_data.js [hunk 8]
```diff
@@ -44,7 +44,7 @@ function shuffleArray(array) {
     return array;
 }
 
-const testBasePrefix = "SIDMTESTWP_" + randomAlphabeticString();
+const testBasePrefix = "iwp_" ;
```
Comment (nikola-naydenov-hmcts) [inline src/test/js/shared/random_data.js:47]: why did we decide to change our usual prefix?

Comment (arun-hmcts) [reply src/test/js/shared/random_data.js:47]: It was mentioned in the jira

**Pull request merged**